docs/design/metadata.md

```diff
@@ -0,0 +1,365 @@
+enable-if = "platform_machine == 'x86_64' or platform_machine == 'AMD64'"
+plugin-api = "provider_variant_x86_64.plugin:X8664Plugin"
+```
+
+### `default-priorities` subtable
+
+The `variant.default-priorities` table contains the [default prority
+information](#default-priorities). It may contain the following keys:
+
+- `namespace` specifying the default namespace priority list. This is
+  a list of strings, where every string must be a namespace name.
+  As indicated in the common metadata section, this key is obligatory.
+
+- `feature` specifying the default feature priority list. This is a list
+  of the canonical feature strings (`namespace :: feature`). This key
+  is optional.
+
+- `property` specifying the default property priority list. This is
+  a list of the canonical property strings
+  (`namespace :: feature :: value`). This key is optional.
+
+### `providers` subtable
+
+The `variant.providers` table contains the [provider information](
+#provider-information). It contains subtables keyed after namespaces,
+and must contain a subtable for every namespace specified
+in `variant.default-priorities.namespace` list.
+
+Every subtable may contain the following keys:
+
+- `plugin-api` that is a string forming the object reference used
+  to load the plugin. It is required.
+
+- `requires` that is a list of strings, where every string is
+  a dependency specifier expressing how to install the provider plugin.
+  It is optional, but it is necessary to support installing plugins
+  in an isolated environment.
+
+- `enable-if` that is a string forming an environment marker,
+  specifying the condition under which the plugin is enabled. It is
+  optional, and the plugin is always enabled if it missing.
+
+
+## Distribution metadata
+
+### Base information
+
+The distribution metadata is defined in the [core metadata
+specifications](
+https://packaging.python.org/en/latest/specifications/core-metadata/).
+The wheel variant format adds headers corresponding to the common
+metadata and to the wheel variant description.
+
+### Example headers
+
+```email
+Metadata-Version: 2.1
+Name: numpy
+Version: 2.2.5
+[...]
+Variant-property: x86_64 :: level :: v3
+Variant-hash: fa7c1393
+Variant-requires: aarch64: provider-variant-aarch64 >=0.0.1,<1; python_version >= '3.9'
+Variant-requires: aarch64: legacy-provider-variant-aarch64 >=0.0.1,<1; python_version < '3.9'
+Variant-enable-if: aarch64: platform_machine == 'aarch64' or 'arm' in platform_machine
+Variant-plugin-api: aarch64: provider_variant_aarch64.plugin:AArch64Plugin
+Variant-requires: x86_64: provider-variant-x86-64 >=0.0.1,<1
+Variant-enable-if: x86_64: platform_machine == 'x86_64' or platform_machine == 'AMD64'
+Variant-plugin-api: x86_64: provider_variant_x86_64.plugin:X8664Plugin
+Variant-default-namespace-priorities: x86_64, aarch64
+Variant-default-feature-priorities: x86_64 :: level
+Variant-default-property-priorities: x86_64 :: avx2 :: on
```

Comment:
nit: For consistency with other core metadata, the keys should use `Dashed-Camel-Case` capitalization

Reply:
Thanks, that makes sense. I was wondering why `variantlib` originally used this case but didn't pursue further and just continued using it for consistency. Since the names are case-insensitive anyway, I'll update the doc immediately. I've also filed https://github.com/wheelnext/variantlib/issues/93 to do it in variantlib.